app/models/user_role.rb

```diff
@@ -1,4 +1,5 @@
 class UserRole < ApplicationRecord
   belongs_to :user
   belongs_to :role
+  belongs_to :store, optional: true
```

Comment:
This is the first time I've seen this. Neat!

Reply:
same for me!